File: jgod/api/schemas/doctrine_v2.py
"""Doctrine Service V2 API Schemas

Pydantic models for Doctrine Management Console API endpoints.
"""

from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel, Field

from jgod.doctrine_v2.models import SectionStatus


class SectionRevisionSchema(BaseModel):
    """Section revision schema"""
    version_id: str
    timestamp: datetime
    operator: str
    change_type: str
    content: Optional[str] = None
    metadata: dict = Field(default_factory=dict)


class DoctrineSectionSchema(BaseModel):
    """Doctrine section schema"""
    section_id: str
    title: str
    current_version_id: str
    draft_version_id: Optional[str] = None
    status: str
    created_at: datetime
    updated_at: datetime
    revision_history: List[SectionRevisionSchema] = Field(default_factory=list)
    source: str = "manual"
    severity: Optional[str] = None
    metadata: dict = Field(default_factory=dict)


class CreateDraftRequest(BaseModel):
    """Request to create a draft"""
    content: str


class DiffResponse(BaseModel):
    """Diff response"""
    diff: str
    from_version_id: str
    to_version_id: str


class BulkActionRequest(BaseModel):
    """Request for bulk actions"""
    section_ids: List[str]
    action: str  # "approve" or "reject"

File: trading-ui/jgod-trading-ui/src/types/doctrineV2.ts
/**
 * Doctrine Service V2 Types
 * 
 * TypeScript type definitions for Doctrine Management Console.
 */

export type SectionStatus = "APPROVED" | "DRAFT" | "PENDING_REVIEW" | "DEPRECATED";

export type ChangeType = "CREATE" | "UPDATE" | "APPROVE" | "REJECT" | "ROLLBACK";

export type SectionRevision = {
  version_id: string;
  timestamp: string;
  operator: string;
  change_type: ChangeType;
  content?: string | null;
  metadata: Record<string, any>;
};

export type DoctrineSection = {
  section_id: string;
  title: string;
  current_version_id: string;
  draft_version_id?: string | null;
  status: SectionStatus;
  created_at: string;
  updated_at: string;
  revision_history: SectionRevision[];
  source: string; // "manual" or "self-repair"
  severity?: string | null;
  metadata: Record<string, any>;
};

export type SectionsResponse = {
  sections: DoctrineSection[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
};

export type VersionContentResponse = {
  section_id: string;
  version_id: string;
  content: string;
};

export type DiffResponse = {
  diff: string;
  from_version_id: string;
  to_version_id: string;
};

